packages/node/src/retry/defaultRetry.ts

```diff
@@ -234,11 +235,13 @@ export class RetryHandler extends BaseRetryHandler {
       await asyncSleep(sleepDuration);
       const isLastTry = numRetries === this._options.retryTimeouts.length;
       const eventsToRetry = eventsBuffer.slice(0, eventCount);
-      const { shouldRetry, shouldReduceEventCount, eventIndicesToRemove } = await this._retryEventsOnce(
+      const { shouldRetry, shouldReduceEventCount, eventIndicesToRemove, response } = await this._retryEventsOnce(
         userId,
         deviceId,
         eventsToRetry,
       );
+      if (this._options.onRetry !== null && response.status !== Status.Success)
```

Comment:
In the interest of covering as many use cases as possible, I don't think this needs to only be called if `response.status !== Status.Success`. The response is part of the callback's arguments, so they can check if the attempt succeeded or failed on their own